lib/cusparse/interfaces.jl

```diff
@@ -79,10 +79,87 @@ for (taga, untaga) in tag_wrappers, (wrapa, transa, unwrapa) in op_wrappers
+    return mul!(similar(CuMatrix{T}, (size(A,1), size(B,2))), A, CuArray(B))
+end
+
+function LinearAlgebra.mul!(C::CuMatrix{T}, A::CuMatrix{T}, B::CuSparseMatrix{T}) where {T}
+    mul!(C, B', A', one(T), zero(T))
+    return C'
+end
+
+Base.:(+)(A::CuSparseMatrixCSR, B::CuSparseMatrixCSR) = geam(one(eltype(A)), A, one(eltype(A)), B, 'O')
+Base.:(-)(A::CuSparseMatrixCSR, B::CuSparseMatrixCSR) = geam(one(eltype(A)), A, -one(eltype(A)), B, 'O')
+
+Base.:(+)(A::CuSparseMatrixCSR, B::Adjoint{T,<:CuSparseMatrixCSR}) where {T} = A + Transpose(conj(B.parent))
+Base.:(-)(A::CuSparseMatrixCSR, B::Adjoint{T,<:CuSparseMatrixCSR}) where {T} = A - Transpose(conj(B.parent))
+Base.:(+)(A::Adjoint{T,<:CuSparseMatrixCSR}, B::CuSparseMatrixCSR) where {T} = Transpose(conj(A.parent)) + B
+Base.:(-)(A::Adjoint{T,<:CuSparseMatrixCSR}, B::CuSparseMatrixCSR) where {T} = Transpose(conj(A.parent)) - B
+Base.:(+)(A::Adjoint{T,<:CuSparseMatrixCSR}, B::Adjoint{T,<:CuSparseMatrixCSR}) where {T} =
+    Transpose(conj(A.parent)) + B
+Base.:(-)(A::Adjoint{T,<:CuSparseMatrixCSR}, B::Adjoint{T,<:CuSparseMatrixCSR}) where {T} =
+    Transpose(conj(A.parent)) - B
+
+function Base.:(+)(A::CuSparseMatrixCSR, B::Transpose{T,<:CuSparseMatrixCSR}) where {T}
+    cscB = CuSparseMatrixCSC(B.parent)
+    transB = CuSparseMatrixCSR(cscB.colPtr, cscB.rowVal, cscB.nzVal, size(cscB))
+    return geam(one(T), A, one(T), transB, 'O')
+end
+
+function Base.:(-)(A::CuSparseMatrixCSR, B::Transpose{T,<:CuSparseMatrixCSR}) where {T}
+    cscB = CuSparseMatrixCSC(B.parent)
+    transB = CuSparseMatrixCSR(cscB.colPtr, cscB.rowVal, cscB.nzVal, size(cscB))
+    return geam(one(T), A, -one(T), transB, 'O')
+end
+
+function Base.:(+)(A::Transpose{T,<:CuSparseMatrixCSR}, B::CuSparseMatrixCSR) where {T}
+    cscA = CuSparseMatrixCSC(A.parent)
+    transA = CuSparseMatrixCSR(cscA.colPtr, cscA.rowVal, cscA.nzVal, size(cscA))
+    geam(one(T), transA, one(T), B, 'O')
+end
+
+function Base.:(-)(A::Transpose{T,<:CuSparseMatrixCSR}, B::CuSparseMatrixCSR) where {T}
+    cscA = CuSparseMatrixCSC(A.parent)
+    transA = CuSparseMatrixCSR(cscA.colPtr, cscA.rowVal, cscA.nzVal, size(cscA))
+    geam(one(T), transA, -one(T), B, 'O')
+end
+
+function Base.:(+)(A::Transpose{T,<:CuSparseMatrixCSR}, B::Transpose{T,<:CuSparseMatrixCSR}) where {T}
+    C = geam(one(T), A.parent, one(T), B.parent, 'O')
+    cscC = CuSparseMatrixCSC(C)
+    return CuSparseMatrixCSR(cscC.colPtr, cscC.rowVal, cscC.nzVal, size(cscC))
+end
+
+function Base.:(-)(A::Transpose{T,<:CuSparseMatrixCSR}, B::Transpose{T,<:CuSparseMatrixCSR}) where {T}
+    C = geam(one(T), A.parent, -one(T), B.parent, 'O')
+    cscC = CuSparseMatrixCSC(C)
+    return CuSparseMatrixCSR(cscC.colPtr, cscC.rowVal, cscC.nzVal, size(cscC))
+end
+
+function Base.:(+)(A::CuSparseMatrixCSR, B::CuSparseMatrix)
+    csrB = CuSparseMatrixCSR(B)
+    return geam(one(eltype(A)), A, one(eltype(A)), csrB, 'O')
+end
```

Comment:
Maybe the `CuSparseMatrixCSR(B)` constructor and when `B` is a `CuSparseMatrixCOO` will trigger the bug.

Reply:
It seems like occur in `cusparseXcoo2csr`.